### ISSUE TITLE
Add stdout trace exporter

### ISSUE THREAD

**Issue by iredelmeier** [open-telemetry/opentelemetry-rust#16]
A stdout exporter (or something that can exporter to a [`Write`r](https://doc.rust-lang.org/std/io/trait.Write.html) can be really valuable, especially for dev purposes.

The [go stdout trace exporter](https://github.com/open-telemetry/opentelemetry-go/tree/e6d725626d4629220a2de0112570adf80d50be21/exporter/trace/stdout) might be useful inspiration.

**Comment by zoidbergwill** [comment]
I took a go at this, but it's gonna take a bit of work for us to better separate Jaeger from the core tracer, then this'll be easier to tackle.

Since the [`sdk::trace::Tracer`](https://github.com/open-telemetry/opentelemetry-rust/blob/master/src/sdk/trace/tracer.rs#L20) relies on a `Jaeger::Tracer`

**Comment by zoidbergwill** [comment]
I think the same will be true for https://github.com/open-telemetry/opentelemetry-rust/issues/17 and the Prometheus stuff.

**Comment by jtescher** [comment]
Clarifying the references between Provider <-> Tracer <-> Span, and how finished spans will get to a processor/exporter would be useful as well. Currently [rustracing_jaeger](https://github.com/sile/rustracing_jaeger) relies on crossbeam-channel and [the examples use a separate thread which listens to messges sent by spans on drop](https://github.com/sile/rustracing_jaeger/blob/master/examples/http_hello_server.rs#L48-L56). 

We should likely have a separate issue where we discuss the best architecture to move forward with as we fill out the current implementation gaps necessary to support additional exporters.

**Comment by jtescher** [comment]
This is now unblocked after #44 landed 🎉

**Comment by jtescher** [comment]
Resolved in #51
